### ISSUE TITLE
No mail notification on talks (discussion mode only)

### ISSUE THREAD

**Issue by simogeo** [nextcloud/spreed#3076]
## Steps to reproduce
1. create channel
2. notify user by explicitly naming him @name 
3. wait, wait, and wait again - no mail is sent

### Expected behaviour
Send email to user

### Actual behaviour
No mail is sent

## Browser

**Microphone available:** no

**Camera available:** no

**Operating system:** Ubuntu

**Browser name:** Firefox

**Browser version:** 50.1/55/...

### Browser log

<details>
```
Insert your browser log here, this could for example include:
a) The javascript console log
b) The network log
c) ...
```

</details>

## Talk app

**Talk app version:** (see apps admin page: `/index.php/settings/apps`)

**Custom TURN server configured:** yes/no (see additional admin settings: `/index.php/settings/admin/additional`)

**Custom STUN server configured:** yes/no (see additional admin settings: `/index.php/settings/admin/additional`)

## Server configuration

**Operating system**: Ubuntu

**Web server:** Apache

**Database:** MySQL

**PHP version:** PHP Version 7.2.24-0ubuntu0.18.04.3

**Nextcloud Version:** 18.0.1

**List of activated apps:**

<details>
```
  - accessibility: 1.4.0
  - activity: 2.11.0
  - bruteforcesettings: 1.5.0
  - calendar: 2.0.2
  - cloud_federation_api: 1.1.0
  - comments: 1.8.0
  - dav: 1.14.0
  - federatedfilesharing: 1.8.0
  - federation: 1.8.0
  - files: 1.13.1
  - files_pdfviewer: 1.7.0
  - files_rightclick: 0.15.2
  - files_sharing: 1.10.1
  - files_trashbin: 1.8.0
  - files_versions: 1.11.0
  - files_videoplayer: 1.7.0
  - firstrunwizard: 2.7.0
  - logreader: 2.3.0
  - lookup_server_connector: 1.6.0
  - nextcloud_announcements: 1.7.0
  - notifications: 2.6.0
  - oauth2: 1.6.0
  - password_policy: 1.8.0
  - photos: 1.0.0
  - privacy: 1.2.0
  - provisioning_api: 1.8.0
  - recommendations: 0.6.0
  - serverinfo: 1.8.0
  - settings: 1.0.0
  - sharebymail: 1.8.0
  - spreed: 8.0.5
  - support: 1.1.0
  - survey_client: 1.6.0
  - systemtags: 1.8.0
  - text: 2.0.0
  - theming: 1.9.0
  - twofactor_backupcodes: 1.7.0
  - updatenotification: 1.8.0
  - viewer: 1.2.0
  - workflowengine: 2.0.0
```
</details>

**Nextcloud configuration:**

<details>
```
{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "localhost",
            "cloud.linea21.com",
            "cloud.intermezzo-coop.eu"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "overwrite.cli.url": "http:\/\/cloud.linea21.com",
        "dbtype": "mysql",
        "version": "18.0.1.3",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "forcessl": true,
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpmode": "sendmail",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "apps_paths": [
            {
                "path": "\/var\/www\/nextcloud\/apps",
                "url": "\/apps",
                "writable": false
            },
            {
                "path": "\/var\/www\/nextcloud\/apps2",
                "url": "\/apps2",
                "writable": true
            }
        ],
        "theme": "",
        "maintenance": false,
        "loglevel": 0,
        "trashbin_retention_obligation": "30,90",
        "versions_retention_obligation": "auto,360",
        "updatechecker": false,
        "updater.release.channel": "stable",
        "mail_sendmailmode": "smtp",
        "updater.secret": "***REMOVED SENSITIVE VALUE***"
    }
}
```
</details>

### Server log (data/nextcloud.log)
<details>
```
[core] Debug: OC\AppFramework\Middleware\Security\Exceptions\CrossSiteRequestForgeryException: CSRF check failed at <<closure>>

0. /var/www/nextcloud/lib/private/AppFramework/Middleware/MiddlewareDispatcher.php line 97
   OC\AppFramework\Middleware\Security\SecurityMiddleware->beforeController(OCA\Text\Controller\SessionController {}, "sync")
1. /var/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php line 98
   OC\AppFramework\Middleware\MiddlewareDispatcher->beforeController(OCA\Text\Controller\SessionController {}, "sync")
2. /var/www/nextcloud/lib/private/AppFramework/App.php line 125
   OC\AppFramework\Http\Dispatcher->dispatch(OCA\Text\Controller\SessionController {}, "sync")
3. /var/www/nextcloud/lib/private/AppFramework/Routing/RouteActionHandler.php line 47
   OC\AppFramework\App::main("OCA\\Text\\Cont ... r", "sync", OC\AppFramework\ ... {}, {_route: "text.Session.sync"})
4. <<closure>>
   OC\AppFramework\Routing\RouteActionHandler->__invoke({_route: "text.Session.sync"})
5. /var/www/nextcloud/lib/private/Route/Router.php line 299
   call_user_func(OC\AppFramework\ ... {}, {_route: "text.Session.sync"})
6. /var/www/nextcloud/lib/base.php line 1008
   OC\Route\Router->match("/apps/text/session/sync")
7. /var/www/nextcloud/index.php line 38
   OC::handleRequest()

POST /index.php/apps/text/session/sync
from 78.194.240.218 by simon at 2020-03-10T21:05:43+00:00
```
</details>

**Comment by nickvergessen** [comment]
Hardly any chat app is sending emails for chat messages.
We send notifications and that should be enough. There are long-term plans on the notification side of things to be able to receive emails for notifications. So we are not going to implement emails in talk.

**Comment by simogeo** [comment]
That's clear enough. Thanks for your message.

**Comment by daxid** [comment]
For discussion sakes :

@nickvergessen you mention in several posts that "hardly any chat app is sending emails".
Well it might be true for "IRC like" apps, but Slack, Rocket chat or Mattermost certainly do.
To me those are a kind of chat app closer to Discussion in Nextcloud than IRC...

I second the demand made in the present issue and [this one](https://github.com/nextcloud/spreed/issues/2919). Without mail notifications, Discussion app is quite unusable in our context (business and academic).

However, the possibility to have that handled  on the notification side of things seems totally reasonable. Too bad it seems not planned in short term. I will certainly fill  a request of improvement on that side. 

Best regards

**Comment by simogeo** [comment]
Indeed, I switched to Slack which notify users by email ....

**Comment by Spartachetto** [comment]
In my humble opinion this shouldn't be a feature of spreed, but an integration between spreed and Nextcloud Flow. 

If the right event is defined for Flow as a trigger, it could be used to send an email. See also https://github.com/nextcloud/mail/issues/2823

**Comment by ma12-co** [comment]
@jancborchardt Insists that this is important, and I can confirm that slack does this by default. Should we continue the discussion?

**Comment by jancborchardt** [comment]
Totally agree there that mail notifications are important – especially if you don’t use Talk that often and/or don’t have the mobile apps installed, which is the case for most people. :)

**Comment by nickvergessen** [comment]
Nextcloud has a notification system and Talk should use it. Nothing more should be necessary. The notification system is missing emails, we are aware, but it is not valued enough to be fixed, so that's how it is.

If we implement another email notification we will have the same problem as with sharing, where it becomes totally confusing when at some point you even get two emails about the same share being done, etc. There is already too many mixups with activity, notifications and the sharing emails. Let's not open the can of worms again in another place.

The issue that should be fixed instead is: https://github.com/nextcloud/notifications/issues/314

**Comment by jospoortvliet** [comment]
> Nextcloud has a notification system and Talk should use it. Nothing more should be necessary. The notification system is missing emails, we are aware, but it is not valued enough to be fixed, so that's how it is.
> 
> If we implement another email notification we will have the same problem as with sharing, where it becomes totally confusing when at some point you even get two emails about the same share being done, etc. There is already too many mixups with activity, notifications and the sharing emails. Let's not open the can of worms again in another place.
> 
> The issue that should be fixed instead is: [nextcloud/notifications#314](https://github.com/nextcloud/notifications/issues/314)

But isn't this what we did with 20? I mean, I can in Activity settings now pick if I want a push or email. I can already get a push or email for getting added to a Chat room, and for receiving a call - perhaps just a extra option there for other Talk events would do the trick...

**Comment by nickvergessen** [comment]
Well we certainly will not push an activity everytime yiu get a chat message. We fixed missing notifications for activity. We didn't fix missing mails for notifications
